CI fails after release.

# Bug reports:

### Expected behavior:

When my PR fails CI in a way that is not reproduced on my machine, it should be made clear how I should fix my PR.

### Current behavior:

PR #2815 was last rebased around Feb 3 or 4.  Squashed yesterday (Feb 9), and CI now fails:
https://github.com/sxs-collaboration/spectre/actions/runs/552845798

The key error message is
```
======================================================================
FAIL: test_get_last_release (tests.tools.Test_CompileReleaseNotes.TestCompileReleaseNotes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/spectre/spectre/tests/tools/Test_CompileReleaseNotes.py", line 25, in test_get_last_release
    self.assertEqual(
AssertionError: 'v2021.02.08' != 'v2021.01.11'
- v2021.02.08
?        ^ ^^
+ v2021.01.11
?        ^ ^^
 : The last release 'v2021.02.08' doesn't match the expected tag name 'v2021.01.11' listed in the repo metadata.

----------------------------------------------------------------------
Ran 4 tests in 0.077s

FAILED (failures=1)
```

I was told I just need to rebase on develop and CI will pass, so I will do that.
But it would be nice if CI succeeds in this case.  If it is really difficult to make CI succeed in cases like this,  at least the error message should say something like "This error may occur if your PR was last rebased before the current release.  In that case, rebasing your PR on develop should fix the problem".
### Environment:

Add as an attachment `$SPECTRE_BUILD_DIR/LibraryVersions.txt` or
add its contents here.

# Feature request:

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [x] Continuous integration

### Desired feature:

- [ ] Detail 1
- [ ] Detail 2
- [ ] Detail 3

### Detailed discussion:

--- a/tools/CompileReleaseNotes.py
+++ b/tools/CompileReleaseNotes.py
@@ -29,11 +29,11 @@ def get_last_release(
         return str(tag).startswith('v')
 
     def distance_from_head(tag):
-        return sum(1 for _ in repo.iter_commits(rev=f'{tag}~1..{head_rev}',
+        return sum(1 for _ in repo.iter_commits(rev=f'{tag}..{head_rev}',
                                                 first_parent=True))
 
     def is_in_history(tag):
-        return distance_from_head(tag) > 0
+        return repo.is_ancestor(ancestor_rev=tag, rev=head_rev)
 
     try:
         return min(filter(is_in_history, filter(is_version_tag, repo.tags)),

